File: src/main/scala/li/cil/oc/common/item/ComponentBus.scala
package li.cil.oc.common.item

import li.cil.oc.Settings
import li.cil.oc.common.item.abstracts.SimpleItem
import net.minecraft.item.Item
import net.minecraft.item.Item.Properties
import net.minecraft.item.ItemStack
import net.minecraftforge.common.extensions.IForgeItem

import scala.jdk.CollectionConverters.SeqHasAsJava

class ComponentBus(props: Properties, val tier: Int) extends SimpleItem(props) with IForgeItem with traits.ItemTier {
  @Deprecated
  override def getDescriptionId = super.getDescriptionId + tier

  override protected def tooltipName = Option(unlocalizedName)

  override protected def tooltipData = Seq(Settings.get.cpuComponentSupport(tier)).map(_.asInstanceOf[AnyRef]).asJava
}

File: src/main/scala/li/cil/oc/common/item/ComponentBus.java
package li.cil.oc.common.item;

import li.cil.oc.Settings;
import li.cil.oc.common.item.abstracts.SimpleItem;
import li.cil.oc.common.item.interfaces.ItemTier;
import net.minecraftforge.common.extensions.IForgeItem;
import scala.Option;

import java.util.Collections;
import java.util.List;
import java.util.Optional;

public class ComponentBus extends SimpleItem implements IForgeItem, ItemTier {
    public final int tier;

    public ComponentBus(Properties props, int tier) {
        super(props);
        this.tier = tier;
    }

    @Override
    public String getDescriptionId() {
        return super.getDescriptionId() + tier;
    }

    @Override
    public Optional<String> tooltipName() {
        return Optional.of(unlocalizedName);
    }

    @Override
    public List<Object> tooltipData() {
        return Collections.singletonList(Settings.get().cpuComponentSupport[tier]);
    }
}
